Extend/Controller/Admin/PaymentMain.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Extend\Controller\Admin;
+
+use OxidEsales\Eshop\Core\Registry;
+use Wirecard\Oxid\Extend\Model\Payment;
+use Wirecard\Oxid\Model\SofortPaymentMethod;
+use Wirecard\PaymentSdk\Config\Config;
+use Wirecard\PaymentSdk\TransactionService;
+
+/**
+ * Controls the payment method config view.
+ *
+ * @mixin \OxidEsales\Eshop\Application\Controller\Admin\PaymentMain
+ */
+class PaymentMain extends PaymentMain_parent
+{
+    /**
+     * @inheritdoc
+     *
+     * @return string
+     */
+    public function render()
+    {
+        $sParentReturn = parent::render();
+
+        if ($this->_isCustomPaymentMethod()) {
+            $sFnc = Registry::getRequest()->getRequestParameter('fnc');
+            $this->_aViewData["bConfigNotValid"] = ($sFnc === 'save' || $sFnc === 'addfield')
+                && !$this->_isSavePossible();
+        }
+
+        return $sParentReturn;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return void
+     */
+    public function save()
+    {
+        if ($this->_isCustomPaymentMethod() && !$this->_isSavePossible()) {
+            $sOxId = $this->getEditObjectId();
+            $this->setEditObjectId($sOxId);
```

Comment:
Why is this get/set required? Can't we just return here?

Reply:
I agree, tested it with and without the two lines and it does not seem to affect the behavior in any way. Will remove it.